File: backend/src/auth/cookies.ts
import type { Response } from "express";

export const SESSION_COOKIE = "sid";

const isProd = process.env.NODE_ENV === "production";
const baseOpts = {
  httpOnly: true,
  secure: isProd,
  sameSite: (isProd ? "none" : "lax") as "none" | "lax",
  path: "/",
};

export function setSessionCookie(res: Response, token: string, expiresAt: Date): void {
  res.cookie(SESSION_COOKIE, token, { ...baseOpts, expires: expiresAt });
}

export function clearSessionCookie(res: Response): void {
  res.clearCookie(SESSION_COOKIE, baseOpts);
}

File: backend-py/app/auth/cookies.py
from datetime import datetime

from fastapi import Response

from ..config import get_settings

SESSION_COOKIE = "sid"
_MAX_AGE = 30 * 24 * 60 * 60


def _opts() -> dict:
    prod = get_settings().is_prod
    return {
        "httponly": True,
        "secure": prod,
        "samesite": "none" if prod else "lax",
        "path": "/",
    }


def set_session_cookie(response: Response, token: str, expires_at: datetime) -> None:
    response.set_cookie(SESSION_COOKIE, token, max_age=_MAX_AGE, **_opts())


def clear_session_cookie(response: Response) -> None:
    response.delete_cookie(SESSION_COOKIE, **_opts())
